Common/Util.cs

```diff
@@ -57,6 +76,66 @@ public static double[] wavelengthsToWavenumbers(double laserWavelengthNM, double
         return wavenumbers;
     }
 
+    public static double wavelengthToWavenumber(double laserWavelengthNM, double wavelength, bool trim = false)
+    {
+        double laserWavenumber = 1.0 / (laserWavelengthNM * NM_TO_CM);
+        if (wavelength > 0)
+        {
+            if (!trim)
+                return laserWavenumber - (1.0 / (wavelength * NM_TO_CM));
+            else
+                return Math.Round(laserWavenumber - (1.0 / (wavelength * NM_TO_CM)), DIGITS_TO_ROUND);
+
+        }
+        else
+            return 0;
+    }
+
+    public static double wavenumberToWavelength(double laserWavelengthNM, double wavenumber, bool trim = false)
+    {
+        if (!trim)
+            return (1.0 / ((1.0 / laserWavelengthNM) - (wavenumber * NM_TO_CM)));
+        else
+            return Math.Round((1.0 / ((1.0 / laserWavelengthNM) - (wavenumber * NM_TO_CM))), DIGITS_TO_ROUND);
+    }
+
+    //
+    // Pearson library match, assumes provided measurements are already corrected to be on the same axis
+    //
+    public static double pearsonLibraryMatch(Measurement sampleM, Measurement library, double airPLSLambda = 10000, int airPLSMaxIter = 100, bool smooth = true)
+    {
+        if (smooth)
+        {
+            logger.info("smoothing sample and library");
+            double[] yIn = AirPLS.smooth(sampleM.processed, airPLSLambda, airPLSMaxIter, 0.001, verbose: false, (int)sampleM.roiStart, (int)sampleM.roiEnd);
+            double[] array = AirPLS.smooth(library.processed, airPLSLambda, airPLSMaxIter, 0.001, verbose: false, (int)sampleM.roiStart, (int)sampleM.roiEnd);
```

Comment:
Does AirPLS actually smooth spectra? I thought it was mainly baseline correction.  I know there's a "WhittakerSmoother" as part of that, but I was thinking that was used to generate the baseline that got subtracted, leaving the resulting pixel-to-pixel noise almost exactly as it started.

Reply:
yeah it's a baseline correction, I don't remember why I went with smooth as opposed to another word